packages/js/components/src/advanced-filters/number-filter.js

```diff
@@ -72,16 +73,11 @@ class NumberFilter extends Component {
 		currencySymbol,
 		symbolPosition,
 	} ) {
+		currencySymbol = decodeEntities( currencySymbol );
```

Comment:
The decoding of HTML entities works well here.

However, if using the example in this PR to set the `symbol` with HTML entities, there are many other places where it won't be rendered properly.

- Such as the Customers table on the same page as in the testing instructions:
   ![1](https://github.com/user-attachments/assets/bbadebb7-b08b-4125-b2d3-5e045b72c64f)
- Most of the analytics pages, like:
   ![image](https://github.com/user-attachments/assets/26f2d459-46d9-48c0-b422-39c0da10130d)

So what I'm trying to figure out is whether the actual use case of allowing HTML entities here would have the same problem? Or is there actually no need to allow HTML entities here?